src/test-utf8.c

```diff
@@ -0,0 +1,43 @@
+/* SPDX-License-Identifier: LGPL-3.0-or-later */
+/** @file test-utf8.c
+ *
+ * @author Juuso Alasuutari
+ */
+#include <errno.h>
+#include <inttypes.h>
+
+#include "dbg.h"
+#include "utf8.h"
+
+int
+main (int    c,
+      char **v)
+{
+	size_t n_codepts = 0;
+	bool separate = false;
+	struct utf8 u8p = utf8();
+
+	for (int i = 0; ++i < c;) {
```

Comment:
Oh come on, the logic of your suggestion is the exact same. And the idea _is_ to skip `argv[0]`.

`for (int i = 0; ++i < argc;)` is an idiom I use because it works well for argv iteration and is concise.